vite.config.ts

```diff
@@ -5,23 +5,62 @@
  * SPDX-License-Identifier: MIT
  */
 import react from "@vitejs/plugin-react";
+import ejs from "ejs";
 import fs from "node:fs";
 import path from "node:path";
-import { UserConfig, defineConfig, loadEnv } from "vite";
-import { configDefaults } from "vitest/config";
+import {
+  IndexHtmlTransformContext,
+  IndexHtmlTransformResult,
+  Plugin,
+  UserConfig,
+  defineConfig,
+  loadEnv,
+} from "vite";
 import svgr from "vite-plugin-svgr";
+import { configDefaults } from "vitest/config";
+
+interface TemplateStrings {
+  appNameFull: string;
+  ogDescription: undefined | string;
+  metaDescription: undefined | string;
+}
 
 // Support optionally pulling in external branding if the module is installed.
 const theme = "@microbit-foundation/ml-trainer-microbit";
 const external = `node_modules/${theme}`;
 const internal = "src/deployment/default";
+const themePackageExternal = fs.existsSync(external);
+const themePackageAlias = themePackageExternal
+  ? theme
+  : path.resolve(__dirname, internal);
+
+const viteEjsPlugin = (data: ejs.Data): Plugin => {
+  return {
+    name: "ejs",
+    transformIndexHtml: {
+      order: "pre",
+      handler: (
+        html: string,
+        _ctx: IndexHtmlTransformContext
+      ): IndexHtmlTransformResult => ejs.render(html, data),
+    },
+  };
+};
 
-export default defineConfig(({ mode }): UserConfig => {
+export default defineConfig(async ({ mode }): Promise<UserConfig> => {
   const commonEnv = loadEnv(mode, process.cwd(), "");
 
+  const strings: TemplateStrings = themePackageExternal
+    ? // eslint-disable-next-line @typescript-eslint/no-unsafe-call, @typescript-eslint/no-unsafe-member-access
+      ((await import(themePackageAlias)).default({}) as TemplateStrings)
+    : {
```

Comment:
Maybe? It's a bit of a compromise as you can import the external theme package but not the one in this project (TypeScript, JSX etc).

Reply:
It's minor additional config for someone forking. Maybe we document this somewhere, but the change looks good and better than the fragile regex solution.